### PR TITLE
Change poe tasks naming

### PR DESCRIPTION
**Pull request by lucekdudek**
Initial gole of this PR was to bump goth's dependencies so it is possible to use `click>=8` (dependency of `mitmproxy` and `black`). Unfortunately, goth yields unexpected errors when using  `mitmproxy>=7`. I ended up with bumping versions slightly. As a result `click>=8` is still not supported.

What has been done:
 - Moved flake8 configuration from `.flake8` to pyproject.toml (this required flak8 bump to `^5`)
 - Added flake8 extend codes to ignore 
 - Bumped mitmproxy to `^6`
 - Unified naming of `tool.poe.tasks` so they match other golemfactory python project (including github workflows configurations)
 - Updated poetry.lock
 - Bump `mypy` to newer version - `poe checks_typing` yields the same warning on both versions 
 - Added separated issue for bumping `mitmproxy` more: https://github.com/golemfactory/goth/issues/605
 
 ---
 
 yapapi compatibility test
 https://github.com/golemfactory/yapapi/pull/1093

---

Edit: After yapapi compatibility test we decided to not upgrade `mitmproxy` at all and only merge changes to poe tasks and dev packages upgrade

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,4 +1,5 @@
 [tool.black]
+line-length = 88
```
Comment (approxit) [inline pyproject.toml:2]: We are using `100` value here.

File: pyproject.toml [hunk 2]
```diff
@@ -40,20 +41,60 @@ ghapi = "^0.1.16"
...
+], help = "Run all available tests" }
+tests_unit = "pytest -svx test/unit"
+tests_integration = "pytest -svx test/integration"
 
 [build-system]
 requires = ["poetry-core>=1.0.0"]
 build-backend = "poetry.core.masonry.api"
+
+[tool.flake8]
+max-line-length = 89
```
Comment (approxit) [inline pyproject.toml:75]: We're using `100` value here.

File: pyproject.toml [hunk 3]
```diff
@@ -40,20 +41,60 @@ ghapi = "^0.1.16"
...
+  "D100", # No docs for public module
+  "D101", # No docs for public class
+  "D102", # No docs for public method
+  "D104", # No docs for public package
+  "D105", # No docs for magic method
+  "D107", # No docs for __init__
+  "D202", # We prefer whitelines after docstrings
+  "D400", # First line should end with a period
+  "D403", # First word of the first line should be properly capitalize
+  "F405", # Star imports
```
Comment (approxit) [inline pyproject.toml:88]: Create separate task to fix D100, D101, D102, D104, D400, D403, F405, and mention it in proper lines.

**Review by approxit — CHANGES_REQUESTED** [review]

**Review by shadeofblue — COMMENTED** [review]
looks okay, but @approxit 's comments still stand

File: pyproject.toml (hunk 3 above)
Comment (lucekdudek) [reply pyproject.toml]: Added
 - https://github.com/golemfactory/goth/issues/606
and
 - https://github.com/golemfactory/goth/issues/607
 
 Linked in the toml

File: goth/runner/download/__init__.py [hunk 4]
```diff
@@ -301,15 +295,13 @@ def download(
         release = self._get_latest_release(tag_substring, content_type, use_unstable)
         if not release:
             raise AssetNotFound(
-                f"Could not find release. "
-                f"repo={self.repo_name}, tag_substring={tag_substring}"
+                f"Could not find release. " f"repo={self.repo_name}, tag_substring={tag_substring}"
```
Comment (approxit) [inline goth/runner/download/__init__.py:298]: This should be a single string.

File: goth/runner/download/__init__.py [hunk 5]
```diff
@@ -301,15 +295,13 @@ def download(
...
             )
 
         asset = self._get_asset(release, content_type, asset_name)
         if not asset:
             raise AssetNotFound(
-                f"Could not find asset. "
-                f"content_type={content_type}, asset_name={asset_name}"
+                f"Could not find asset. " f"content_type={content_type}, asset_name={asset_name}"
```
Comment (approxit) [inline goth/runner/download/__init__.py:304]: This should be a single string.

File: goth/runner/log_monitor.py [hunk 6]
```diff
@@ -140,8 +136,7 @@ async def wait_for_pattern(
             return event
         except asyncio.TimeoutError:
             raise goth_exceptions.TimeoutError(
-                f"Log monitor timed out waiting for event. "
-                f"pattern={pattern}, timeout={timeout}"
+                f"Log monitor timed out waiting for event. " f"pattern={pattern}, timeout={timeout}"
```
Comment (approxit) [inline goth/runner/log_monitor.py:139]: This should be a single string.

File: goth/runner/probe/__init__.py [hunk 7]
```diff
@@ -251,14 +251,11 @@ async def _start_container(self) -> None:
...
-        self.ip_address = get_container_address(
-            self._docker_client, self.container.name
-        )
+        self.ip_address = get_container_address(self._docker_client, self.container.name)
         nginx_ip_address = self.runner.nginx_container_address
 
         self._logger.info(
-            "Yagna API host:port in Docker network: "
-            "%s:%s (direct), %s:%s (through proxy)",
+            "Yagna API host:port in Docker network: " "%s:%s (direct), %s:%s (through proxy)",
```
Comment (approxit) [inline goth/runner/probe/__init__.py:258]: This should be a single string.

File: goth/runner/probe/__init__.py [hunk 8]
```diff
@@ -356,9 +351,7 @@ async def run_command_on_host(
         command: str,
         env: Optional[Mapping[str, str]] = None,
         command_timeout: float = 300,
-    ) -> AsyncIterator[
-        Tuple[asyncio.Task, PatternMatchingEventMonitor, process.ProcessMonitor],
-    ]:
+    ) -> AsyncIterator[Tuple[asyncio.Task, PatternMatchingEventMonitor, process.ProcessMonitor],]:
```
Comment (approxit) [inline goth/runner/probe/__init__.py:354]: Trailing `,` is not needed here.

File: pyproject.toml [hunk 9]
```diff
@@ -40,20 +41,60 @@ ghapi = "^0.1.16"
...
 [build-system]
 requires = ["poetry-core>=1.0.0"]
 build-backend = "poetry.core.masonry.api"
+
+[tool.flake8]
+max-line-length = 100
+extend-ignore = [
+  "E203", # See https://github.com/PyCQA/pycodestyle/issues/373
+  "E231", # black ignores this rule when formatting
+  "D100", # No docs for public module https://github.com/golemfactory/goth/issues/606
```
Comment (approxit) [inline pyproject.toml:79]: Mention intent of linked issue, aka `to be fixed by`.

**Review by approxit — CHANGES_REQUESTED** [review]
Also I can see `poetry.lock` in repo, which currently is not included in VCS across SDK libraries.

File: pyproject.toml [hunk 10]
```diff
@@ -40,20 +41,60 @@ ghapi = "^0.1.16"
...
-unit_test = "pytest -svx test/unit"
-integration_test = "pytest -svx test/integration"
+checks_codestyle = { sequence = [
+  "_checks_codestyle_flake8",
+  "_checks_codestyle_black",
+], help = "Run only code style checks" }
+_checks_codestyle_flake8 = "flake8"
+_checks_codestyle_black = "black --check --diff ."
+checks_typing = { cmd = "mypy --install-types --non-interactive --ignore-missing-imports --check-untyped-defs --warn-unused-ignores --show-error-codes .", help = "Run only code typing checks" }
+format = { sequence = ["_format_black"], help = "Run code auto formatting" }
```
[the remark this answers is not in the document]
Comment (lucekdudek) [reply pyproject.toml:61]: I am not going to do this in this PR

**Review by shadeofblue — APPROVED** [review]
lgtm

**Comment by lucekdudek** [comment]
> Also I can see `poetry.lock` in repo, which currently is not included in VCS across SDK libraries.

It always has .lock so we can keep it as for now

**Review by approxit — APPROVED** [review]

**Pull request merged**